File: quantask-web/src/app/core/services/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface Comment {
    id: number;
    task_id: number;
    user_id: number;
    content: string;
    created_at: string;
    user?: {
        id: number;
        name: string;
        avatar_url?: string;
    };
}

@Injectable({
    providedIn: 'root'
})
export class CommentService {
    private apiUrl = environment.apiUrl;

    constructor(private http: HttpClient) { }

    // Get comments for a task
    getComments(taskId: number): Observable<Comment[]> {
        return this.http.get<Comment[]>(`${this.apiUrl}/tasks/${taskId}/comments`);
    }

    // Create comment
    createComment(taskId: number, content: string): Observable<Comment> {
        return this.http.post<Comment>(`${this.apiUrl}/tasks/${taskId}/comments`, { content });
    }

    // Delete comment
    deleteComment(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/comments/${id}`);
    }
}

File: quantask-api/features/task/services/comment_service.rb
module Task
  module Services
    class CommentService
      def self.create(user, params)
        # Merge task_id from params if present
        comment_params = params[:task_id] ? params : params.merge(task_id: params[:task_id])
        ::Task::Commands::CreateComment.call(user, comment_params)
      end

      def self.delete(user, id)
        ::Task::Commands::DeleteComment.call(user, id)
      end
    end
  end
end
